### PR TITLE
FSharpTreeBuilderBase/TypingAssist: fix for constrained return type usage

### PR DESCRIPTION
**Pull request by DedSec256**

### CODE REVIEW

File: ReSharper.FSharp/test/data/parsing/Type member - Abstract 04 - Constrained type usage.fs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module Module
+
+type T =
+    abstract member Using: resource:'T * binder:('T -> Async<'U>) -> Async<'U> when 'T :> System.IDisposable
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/parsing/Type member - Abstract 04 - Constrained type usage.fs:4]: @DedSec256 Could you please simplify the test, so it's easier to see what exactly being tested here?

File: ReSharper.FSharp/test/data/parsing/signatures/Type member - Constrained type usage.fsi [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module Module
+
+type T =
+    member Using: resource:'T * binder:('T -> Async<'U>) -> Async<'U> when 'T :> System.IDisposable
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/parsing/signatures/Type member - Constrained type usage.fsi:4]: Likewise.

**Review by auduchinok — CHANGES_REQUESTED** [review]

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpTreeBuilderBase.fs [hunk 3]
```diff
@@ -678,6 +678,11 @@ type FSharpTreeBuilderBase(lexer, document: IDocument, lifetime, path: VirtualFi
 
         let rec loop paramGroups returnInfo synType =
             match paramGroups, synType with
+            | _, SynType.WithGlobalConstraints(synType, constraints, range) ->
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpTreeBuilderBase.fs:681]: Is the loop over parameter groups an appropriate place to add this change to? Can these constraints go after each parameter group?

Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpTreeBuilderBase.fs:681]: no, only at the end of the signature, possibly combined via 'and'

**Review by auduchinok — APPROVED** [review]
@DedSec256 Thank you for this

**Pull request merged**